src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -135,6 +135,7 @@ namespace cryptonote
     std::atomic<bool> m_synchronized;
     bool m_one_request = true;
     std::atomic<bool> m_stopping;
+    epee::critical_section m_sync_lock;
```

Comment:
I used this because the rest of the code uses this. Though I did not know about the sleep. This strikes me a bit wtf, but I guess I'll go look at the code later to see what this is about.